src/imu/mpu9255/mpu9255.cpp

```diff
@@ -0,0 +1,150 @@
+#include "mpu9255.hpp"
+
+namespace imu {
+
+auto Mpu9255::Init(void) noexcept -> types::DriverStatus {
+  CreateSensorPointer();
+
+  SetInitConfigMPU9255();
+
+  if (InitAllSensors()) {
+    initialized_ = true;
+    return types::DriverStatus::OK;
+  }
+  return types::DriverStatus::HAL_ERROR;
+}
+
+auto Mpu9255::CreateSensorPointer(void) noexcept -> void {
+  if (gyroscope_ == NULL) {
+    gyroscope_ = std::make_unique<imu::Gyroscope>(i2c_handler_);
+  }
+  if (accelerometer_ == NULL) {
+    accelerometer_ = std::make_unique<imu::Accelerometer>(i2c_handler_);
+  }
+  if (magnetometer_ == NULL) {
+    magnetometer_ = std::make_unique<imu::Magnetometer>(i2c_handler_);
+  }
+  if (temperature_ == NULL) {
+    temperature_ = std::make_unique<imu::Temperature>(i2c_handler_);
+  }
+}
+
+auto Mpu9255::SetInitConfigMPU9255(void) noexcept -> void {
+  SetMPU9255Register(INT_PIN_CFG, 0x22);
```

Comment:
Magic number, at least give it a name inside of this method. You can make a constexpr variable inside this function and let the compiler distinguish if it is true constexpr or only const.

Reply:
Used utilities::Byte and comment to declare meaning of bits. Good hint, much better now!